### DISCUSSION TITLE
Adding Description and Comments in Superposition

### DISCUSSION THREAD

**Discussion by sauraww** [juspay/superposition#292] [Ideas]
For experiments, dimensions, configs, and overrides in Context-Aware Configuration System , Add Description/Commit Message Columns.

`Description and Comment`

Add the description and comment column in all the existing tables

`Contexts and Overrides `

Add comment in the config_versions table. 
Use config_versions to get the history of all previous contexts and overrides.
 For Batch API Operations: Use a single commit message at the end of the batch operation (snapshot). 
This solution leverages the existing config_versions mechanism to track changes and associate commit messages
 with snapshots of the entire configuration.

`History Tables `

Add history tables for Default Config , Dimension , Experiment ( experiment ramp , etc )

**Comment by ayushjain17** [comment]
### Point 1:

Currently in the PR, `comment` column is added in all tables

I am not sure if `comment` as a column makes sense in all the tables when the data stored in the tables are not transactional. We will always be overwriting the `comment` column with the new data whenever we are making a new change and giving a new comment, as we are updating the same row itself, which makes it behave exactly same as the `description` column. 

`comment` as a column only makes sense if we add it to AuditLog/EventLog table, where we are storing things as a transaction and then we can store the comment as a form of reason for that action (change-log) in particular.

Here, having `comment` as mandatory makes sense for both the db type as well in the APIs.

### Point 2:
In the PR, `description` has been made mandatory everywhere, be it all the API requests or DB column.

Even if we want to keep `description` as mandatory, it should be mandatory in the db type and while creating an entity only and not while updating it. 

While updating, if the user forgets to give the input for `description` this would simply overwrite the column and make it empty, which is incorrect. 
It should not be needed again and again, as the `description` field serves to give a meaning to the existence of an entry in the db in words, this would not be intend to be updated every-time while making a change.

**Comment by pratikmishra356** [comment]
https://github.com/juspay/superposition/issues/354
@knutties should we remove description column from snapshot/config_version table ?

**Reply by knutties** [reply]
sounds ok to me.
